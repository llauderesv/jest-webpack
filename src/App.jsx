import React from 'react';

import './App.scss';

function App() {
  return (
    <div className="app">
      <div className="header">
        <div className="header-wrapper">
          <h1>Welcome to React starter kit.</h1>
          <h3>A simple React setup using Webpack with Jest integration. </h3>
          <div className="build-status">
            <a href="https://github.com/llauderesv/react-starter-kit">
              <img
                src="https://travis-ci.org/llauderesv/react-starter-kit.svg?branch=master"
                alt="travis-ci-build"
              />
            </a>
          </div>
          <small>
            For more info, follow me on twitter
            <a href="https://twitter.com/llaudevc">llaudevc</a>.
          </small>
        </div>
      </div>

      <div className="documentation">
        <h2>Steps to run the project.</h2>

        <div className="code-container">
          <p>Running development mode.</p>
          <pre>npm run dev</pre>
        </div>

        <div className="code-container">
          <p>Running production mode.</p>
          <pre>npm run dev</pre>
        </div>

        <div className="code-container">
          <p>Running unit test.</p>
          <pre>npm run test</pre>
        </div>
      </div>
    </div>
  );
}

export default App;
