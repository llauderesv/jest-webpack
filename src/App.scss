@import './assets/styles/fonts'; // import fonts
@import './assets/styles/baseColor'; // import baseColors

html {
  font-family: 'Proxima';
}

body {
  position: absolute;
  margin: 0;
  padding: 0;
  height: 100%;
  width: 100%;
}

.header {
  width: 100%;
  background-color: #3c3c3cfa;
  height: 300px;
  color: rgb(216, 216, 216);

  .header-container {
    text-align: center;
    font-weight: 500;
    position: relative;
    top: 50%;
    transform: translateY(-50%);

    h1 {
      margin: 0 0 20px 0;
    }

    a {
      margin-left: 5px;
      color: $primary-color;
    }
  }

  .build-status {
    margin: 10px;
  }
}

.steps {
  position: absolute;
  width: 40%;
  left: 50%;
  transform: translateX(-50%);

  .code-container {
    pre {
      background-color: rgb(238, 238, 238);
      padding: 20px;
    }
  }
}
