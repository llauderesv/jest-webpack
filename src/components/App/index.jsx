import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import {
  BrowserRouter as Router,
  Route,
  Switch,
  HashRouter,
} from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import { connect } from 'react-redux';
import routes from '../../routes';

// Component section
import Navbar from '../Navbar';
import ErrorMessage from '../ErrorMessage';

import './index.scss';

const ErrorMessageWithConnect = connect(
  state => ({ message: state.error.message }),
  dispatch => ({
    hideError: () => dispatch({ type: 'HIDE_ERROR' }),
  }),
)(ErrorMessage);

const App = ({ browserHistory, isShow }) => {
  const routesArray = routes.map((route, index) => (
    <Route key={index} {...route} />
  ));

  return (
    <ConnectedRouter history={browserHistory}>
      <HashRouter>
        <Fragment>
          <div className="app-content">
            <Navbar />
            {isShow && <ErrorMessageWithConnect dismissible />}
            <Switch>{routesArray}</Switch>
          </div>
        </Fragment>
      </HashRouter>
    </ConnectedRouter>
  );
};

App.propTypes = {
  browserHistory: PropTypes.object.isRequired,
  isShow: PropTypes.bool.isRequired,
};

export default connect(state => ({ isShow: state.error.isShow }))(App);
