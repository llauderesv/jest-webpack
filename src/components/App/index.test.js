import React from 'react';
import TestRenderer from 'react-test-renderer';
import App from './App';

// eslint-disable-next-line no-undef
describe('App Component', () => {
  it('Should render w/out crashing', () => {
    const app = TestRenderer.create(<App />);

    const tree = app.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
