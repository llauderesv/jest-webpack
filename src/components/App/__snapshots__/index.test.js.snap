// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Should render w/out crashing 1`] = `
Array [
  <nav
    className="navbar navbar-expand navbar-dark bg-dark sticky-top"
  >
    <a
      aria-current={null}
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      IS APAC Template
    </a>
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        aria-current={null}
        className="nav-link"
        href="/"
        onClick={[Function]}
        role="button"
      >
        Home
      </a>
      <a
        aria-current={null}
        className="nav-link"
        href="/examples"
        onClick={[Function]}
        role="button"
      >
        Examples
      </a>
      <a
        aria-current={null}
        className="nav-link"
        href="/about"
        onClick={[Function]}
        role="button"
      >
        About
      </a>
    </div>
  </nav>,
  <div
    className="app-content"
  >
    <div
      className="jumbotron"
    >
      <div
        className="wrapper"
      >
        <h1>
          Welcome to new IS APAC Template
        </h1>
        <p>
          This new template was built using react + redux and integration of other libraries such as react router for handling Single Page Application and redux saga for handling side effects and react bootstrap for styling.
        </p>
        <button
          className="btn btn-secondary"
          disabled={false}
          type="button"
        >
          Learn more
        </button>
      </div>
    </div>
  </div>,
]
`;
